proxy/envoy/watcher.go

```diff
@@ -38,46 +43,47 @@ type watcher struct {
 
 // NewWatcher creates a new watcher instance with an agent
 func NewWatcher(discovery model.ServiceDiscovery, ctl model.Controller,
-	registry *model.IstioRegistry, mesh *MeshConfig) (Watcher, error) {
-	addrs, err := hostIP()
-	glog.V(2).Infof("host IPs: %v", addrs)
-	if err != nil {
-		return nil, err
+	registry *model.IstioRegistry, mesh *MeshConfig, identity *ProxyNode) (Watcher, error) {
+	addrs := make(map[string]bool)
+	if identity.IP != "" {
```

Comment:
it means there is no co-located service instance (e.g. no localhost routes) which is OK.
this could happen for an ingress proxy.